### DISCUSSION TITLE
Pointer of a static lambda

### DISCUSSION THREAD

**Discussion by WhiteBlackGoose** [dotnet/csharplang#4791] [Language Ideas] [answered]
So, there are static lambdas, which is great syntax sugar for inlining methods which you don't want to use twice and passing them as lambdas.

There are also normal static methods, to which we can get a function pointer for high-performance/PInvoke and anything.

So... why not to have pointers to static lambdas?

The syntax might look like
```cs
&static a => 5
```
The type inference will work based on the target pointer, so e. g.
```cs
delegate*<int, int> inc = &static a => a + 1;
```
This
```cs
var inc = &static a => a + 1;
```
is invalid (same way as with normal lambdas).

<hr>

*I guess* the code 
```cs
delegate*<int, int> inc = &static a => a + 1;
```
will be expanded by Roslyn into
```cs
delegate*<int, int> inc = &SomethingQuack.Method;

...

static class SomethingQuack
{
    internal static int Method(int a) => a + 1;
}
```

So, what do you think?

**Comment by 333fred** [comment, marked as answer]
https://github.com/dotnet/csharplang/issues/3476

**Reply by WhiteBlackGoose** [reply]
Oops, thanks
